### ISSUE TITLE
Relation between softmax and cross-entropy

### ISSUE THREAD

**Issue by pakozm** [april-org/april-ann#5]
It is important to review the softmax and cross-entropy. Currently, the derivative of the softmax is asuming cross-entropy error function, and cross-entropy is asuming softmax activation. The use of one without the other is not correct.
